framework/math/math.h

```diff
@@ -99,11 +99,54 @@ void PrintVector(const std::vector<double>& x);
 /// Scales a vector in place by constant.
 void Scale(std::vector<double>& x, const double& val);
 
+/// Scale the vector with a constant value
+template <typename TYPE>
+void
+Scale(Vector<TYPE>& a, TYPE alpha)
+{
+  for (unsigned int i = 0; i < a.Rows(); i++)
+    a(i) *= alpha;
+}
+
+template <typename TYPE>
+Vector<TYPE>
+Scaled(const Vector<TYPE>& a, TYPE alpha)
```

Comment:
Why is this routine called `Scaled` and not `Scale`?

Reply:
Because it returns scaled version of the vector. `Scale` scales the vector, i.e. changes the parameter. I'm following the pattern that is in the code for `Normalize`/`Normalized`.